lessons/git/intro/slides-2017.Rmd

```diff
@@ -0,0 +1,300 @@
+---
+title: "Intro to Git"
+author: "Shyam Srinivasan"
+date: "October 5, 2017"
+output: beamer_presentation
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+# Learning expectations
+
+This won't even touch the surface of git... But! Hopefully, you:
+
+- Recognize the power of using version control
+- Know the basic tools to get started using git
+- Where to go for help
+
+But! No expectation to actually start using it :)
+
+# What is Git?
+![](https://imgs.xkcd.com/comics/git.png)
+
+# Getting started with Git and GitHub
+
+Create an account on GitHub
+
+* You set your username, email and password here
+
+Install and run Git and GitHub
+
+https://desktop.github.com/
+
+* Make sure you install posh-git along with GitHub
+
+# Configuring your setup:
+
+```bash
+git config --global user.name "Your Name"
+git config --global user.email "you@some.domain"
+```
+
+Optional Configurations
+```bash
+git config --global color.ui "auto" 
+```
+
+Setup Editor for Text Files
+
+```bash
+git config --global core.editor "your_editor"
+git config --list # To confirm
+```
+
+# Creating a Repo on Github
+The cloud (GitHub) serves as an online repo where files are accessible 24x7
+
+* Go to [GitHub](https://github.com/)
+
+* Create a new repo by pressing the `New Repository` button
+
+* Let us keep the name short and self explanatory and call it 'git-training' 
+
+* We will call this the `remote` repo from here on
+
+# Creating a Git Repo
+* change directory (move to another folder) using 
+```bash 
+cd
+```
+* to move back one folder use
+```bash 
+cd ..
+```
+* to make a new directory
+```bash
+mkdir
+```
+
+# Let us create a `local` Repo
+```bash
+cd ~/Documents
+# for windows users with full path
+cd C:/Users/user_name/Documents
+# create a new folder 
+# use the same name as your remote repo on GitHub
+mkdir git-training
+# move to the new folder
+cd git-training
+```
+## Convert folder to **git** repo
+```bash
+git init
+```
+## check your folder to find git configuration files/folders
+This folder is now a git repo (short for repository)
+
+# Using GitHub with Git
+If you have GitHub (on Windows/MacOS), you can add the repository you cloned/created to it and visualize the changes
+
+Remember, you installed this earlier?
+
+# Configuring your repo for use with GitHub on the cloud
+In order to use this local repo with the one you created earlier we need to set the remote configuration
+```bash
+git remote -v
+```
+will tell you what is the remote corresponding to your local repo
+
+If it is empty, you can setup a remote using 
+```bash
+git remote add remote_name remote_url
+```
+
+Before you do this, you need to have an account on GitHub and a repo on it
+
+# Configuring your repo for use with GitHub on the cloud
+* Copy the repo url from the repo creation page and use it in
+```bash
+git remote add origin 
+https://github.com/shyamsrinivasan/git-training.git
```

Comment:
'git-training' or 'sandbox-test'.

Reply:
Both. git-training for creation of local repo and sandbox-test for cloning

Reply:
Okay sounds fine then.